include/boost/math/ccmath/signbit.hpp

```diff
@@ -7,19 +7,173 @@
 #define BOOST_MATH_CCMATH_SIGNBIT_HPP
 
 #include <cmath>
+#include <cstdint>
 #include <limits>
 #include <type_traits>
 #include <boost/math/tools/is_constant_evaluated.hpp>
+#include <boost/math/special_functions/detail/fp_traits.hpp>
 #include <boost/math/ccmath/isnan.hpp>
+#include <boost/math/ccmath/abs.hpp>
+
+#if __cpp_lib_bit_cast >= 201806L
+#include <bit>
+#  define BOOST_MATH_BIT_CAST(T, x) std::bit_cast<T>(x)
+#elif defined(__has_builtin)
+#  if __has_builtin(__builtin_bit_cast)
+#    define BOOST_MATH_BIT_CAST(T, x) __builtin_bit_cast(T, x)
+#  endif
+#endif
+
+/*
+The following error is given using Apple Clang version 13.1.6
+TODO: Remove the following undef when Clang supports
+
+ccmath_signbit_test.cpp:32:19: error: static_assert expression is not an integral constant expression
+    static_assert(boost::math::ccmath::signbit(T(-1)) == true);
+                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+../../../boost/math/ccmath/signbit.hpp:62:24: note: constexpr bit_cast involving bit-field is not yet supported
+        const auto u = BOOST_MATH_BIT_CAST(float_bits, arg);
+                       ^
+../../../boost/math/ccmath/signbit.hpp:20:37: note: expanded from macro 'BOOST_MATH_BIT_CAST'
+#  define BOOST_MATH_BIT_CAST(T, x) __builtin_bit_cast(T, x)
+                                    ^
+*/
+
+#if defined(__clang__) && defined(BOOST_MATH_BIT_CAST)
```

Comment:
Is this all clang's or just Apples?

There is an alternative implementation which avoids bit fields: just store the left shift which puts the sign bit at index 0, then bit_cast to an appropriately sized integer, shift and bitwise & with 1u.

Reply:
I just ran clang 12,13, and 14 on Ubuntu. Looks like strictly an Apple problem.